common/djangoapps/embargo/tests/test_api.py

```diff
@@ -0,0 +1,399 @@
+        if profile_country in ["CA", "AF", "IR"]:
+            embargo_url = reverse('embargo')
+            self.assertRedirects(response, embargo_url)
+            self.assertEqual(response.status_code, 302)
+        else:
+            self.assertEqual(response.status_code, 200)
+
+    @mock.patch.dict(settings.FEATURES, {'ENABLE_COUNTRY_ACCESS': True})
+    @ddt.data("", "US", "CA", "NZ", "IR", "CU")
+    def test_embargo_course_blacklisted_with_profile_country(self, profile_country):
+        # if course is emabargoed and has black list countries ( CU , IR ).
+        # then users from these countries can't access this course.
+        # any user from other than these countries can access.
+
+        profile = self.user.profile
+        profile.country = profile_country
+        profile.save()
+
+        response = self.client.get(self.embargoed_course_blacklisted)
+        if profile_country in ["", "US", "CA", "NZ"]:
+            self.assertEqual(response.status_code, 200)
+        else:
+            embargo_url = reverse('embargo')
+            self.assertRedirects(response, embargo_url)
+            self.assertEqual(response.status_code, 302)
+
+    @mock.patch.dict(settings.FEATURES, {'ENABLE_COUNTRY_ACCESS': True})
+    @ddt.data("", "US", "CA", "NZ", "IR", "CU")
+    def test_embargo_course_without_whitelist(self, profile_country):
+        # if course is emabargoed but without black or whitelist
+        # then course can be accessible from any where
+
+        profile = self.user.profile
+        profile.country = profile_country
+        profile.save()
+
+        embargo_course3 = CourseFactory.create()
+        embargo_course3.save()
+        RestrictedCourse(course_key=embargo_course3.id).save()
+        embargoed_course_page = '/courses/' + embargo_course3.id.to_deprecated_string() + '/info'
+
+        response = self.client.get(embargoed_course_page)
+        self.assertEqual(response.status_code, 200)
+
+    @mock.patch.dict(settings.FEATURES, {'ENABLE_COUNTRY_ACCESS': True})
+    @ddt.data("", "US", "CA", "NZ", "IR", "CU")
+    def test_embargo_course_without_blacklist(self, profile_country):
+        # if course is emabargoed but without black or whitelist
+        # then course can be accessible from any where
+        profile = self.user.profile
+        profile.country = profile_country
+        profile.save()
+
+        embargo_course4 = CourseFactory.create()
+        embargo_course4.save()
+        RestrictedCourse(course_key=embargo_course4.id).save()
+        embargoed_course_page = '/courses/' + embargo_course4.id.to_deprecated_string() + '/info'
+
+        response = self.client.get(embargoed_course_page)
+        self.assertEqual(response.status_code, 200)
+
+
+    @mock.patch.dict(settings.FEATURES, {'ENABLE_COUNTRY_ACCESS': True})
+    def test_embargo_profile_country_cache(self):
+        # Warm the cache
+        with self.assertNumQueries(25):
```

Comment:
I'm really surprised that the number of queries here is so high.  Are these coming from the embargo middleware, or from the courseware page?  (One way to tell is to use the Django debug toolbar in devstack -- it will show you which SQL queries are executing).